src/entity/SquadPostModeration.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post, PostType } from './posts';
+import { Source } from './Source';
+
+export enum SquadPostModerationStatus {
+  Approved = 'approved',
+  Rejected = 'rejected',
+  Pending = 'pending',
+}
+
+@Entity()
+export class SquadPostModeration {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @Column({ type: 'text' })
+  sourceId: string;
+
+  @ManyToOne('Source', (source: Source) => source.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  source: Promise<Source>;
+
+  @Column({
+    type: 'enum',
+    enum: SquadPostModerationStatus,
+    default: SquadPostModerationStatus.Pending,
+  })
+  status: SquadPostModerationStatus;
+
+  @Column({ type: 'text' })
+  createdById: string;
+
+  @Column({ type: 'text', nullable: true })
+  moderatedById: string | null;
```

Comment:
We should create the FK indexes for these two columns similar to the `sourceId`.

Reply:
@sshanzel I thought so too, but the specs didn't mention it, and I looked at some other entities we have, and there's plenty of these references to other entities that don't have explicit FK's. Is there a specific reason for that? 🤔

Reply:
Yeah, the previous ones were just missed, but as much as we can, it is better to have them FKs defined for faster query. Especially we will be filtering these per user for their requested posts.

Reply:
@sshanzel Gotchu. I thought there was maybe a reason for it, so since it wasn't specified, I didn't apply it either. I'll fix it up! 😄